lib/modules/dosomething/dosomething_reportback/dosomething_reportback.module

```diff
@@ -959,16 +1003,30 @@ function dosomething_reportback_get_reportback_files_query($params = array()) {
   $query->join('users', 'u', 'rb.uid = u.uid');
   $query->join('node', 'n', 'rb.nid = n.nid');
   $query->join('file_managed', 'f', 'rbf.fid = f.fid');
+
   if (isset($params['tid'])) {
     $query->join('field_data_field_primary_cause', 't', 't.entity_id = n.nid');
     $query->condition('t.field_primary_cause_tid', $params['tid']);
     $query->condition('t.entity_type', 'node');
   }
+
   if (isset($params['status'])) {
     $query->condition('rbf.status', $params['status']);
   }
+
   if (isset($params['nid'])) {
-    $query->condition('rb.nid', $params['nid']);
+    if (is_array($params['nid'])) {
+      $query->condition('rb.nid', $params['nid'], 'IN');
+    } else {
+      $query->condition('rb.nid', $params['nid']);
```

Comment:
Actually, I just need to modify the `formatData()` Transformer method cause it's already doing some stuff in there.... the tricky part is that we can now pass a single int as `?campaigns=362` or a series of ints as `?campaigns=362,1301,293`... The `formatData()` methods takes these comma separated values and turns them into an array of values that then get passed into the query. So to validate, I'd need to check if each of these are actually "numbers" if wanting numbers otherwise toss it out... Maybe have to think it through a bit.